Library/Homebrew/cask/lib/hbc/cli/internal_stanza.rb

```diff
@@ -68,10 +71,16 @@ def run
             next
           end
 
-          if artifact_name && !value.key?(artifact_name)
-            opoo "no such stanza '#{artifact_name}' on Cask '#{cask}'" unless quiet?
-            puts ""
-            next
+          if (value.nil? || value.is_a?(Array) && value.empty?) ||
+             (artifact_name && !value.key?(artifact_name))
```

Comment:
I think the `value.key?(artifact_name)` won't work anymore, since `cask.artifacts` now returns a `Set` of artifacts instead of a `Hash` of `name => [artifacts]`. So to check if there is an artifact with a given name, the logic has to be changed.

Reply:
I have a test case for this that seems to work fine ...